test/cases/sizeof_and_typeof.zig

```diff
@@ -7,28 +7,44 @@ test "sizeofAndTypeOf" {
 const x: u16 = 13;
 const z: @typeOf(x) = 19;
 
-const A = struct {
+const S = struct {
     a: u8,
     b: u32,
     c: u8,
+    d: u3,
+    e: u5,
+    f: u16,
+    g: u16,
 };
 
 const P = packed struct {
     a: u8,
     b: u32,
     c: u8,
+    d: u3,
+    e: u5,
+    f: u16,
+    g: u16,
 };
 
-test "offsetOf" {
-    // Packed structs have fixed memory layout
+test "byteOffsetOf" {
     const p: P = undefined;
-    assert(@offsetOf(P, "a") == 0);
-    assert(@offsetOf(@typeOf(p), "b") == 1);
-    assert(@offsetOf(@typeOf(p), "c") == 5);
-
-    // Non-packed struct fields can be moved/padded
-    const a: A = undefined;
-    assert(@ptrToInt(&a.a) - @ptrToInt(&a) == @offsetOf(A, "a"));
-    assert(@ptrToInt(&a.b) - @ptrToInt(&a) == @offsetOf(@typeOf(a), "b"));
-    assert(@ptrToInt(&a.c) - @ptrToInt(&a) == @offsetOf(@typeOf(a), "c"));
+    std.debug.assert(@byteOffsetOf(P, "a") == 0 and @byteOffsetOf(S, "a") == 0);
```

Comment:
I thought the compiler could reorder fields for regular `struct`s, so it seems to me the bit and byte offset tests using S might be incorrect depending changes in the compiler or settings such as the target. The original tests using the difference of addresses seems like a better assertation.